config/config_partial.go

```diff
@@ -99,6 +99,16 @@ func DecodeBaseBlocks(
 	filename string,
 	includeFromChild *IncludeConfig,
 ) (*cty.Value, *terragruntInclude, TrackInclude, error) {
+	// first decode of IamRole which can be referenced in other blocks
+	contextExtensions := EvalContextExtensions{}
+	flags := terragruntFlags{}
+	err := decodeHcl(hclFile, filename, &flags, terragruntOptions, contextExtensions)
+	if err != nil {
+		return nil, nil, TrackInclude{}, err
+	}
+	if flags.IamRole != nil {
+		terragruntOptions.IamRole = *flags.IamRole
+	}
```

Comment:
I think this should check if `terragruntOptions.IamRole` is already set or not, since we should prefer to use the CLI passed in option over the local configuration.